### DISCUSSION TITLE
Display debug/error info while deployed to server

### DISCUSSION THREAD

**Discussion by JoeFerrucci** [vercel/next.js#45414] [Help]
<img width="996" alt="Screen Shot 2023-01-30 at 12 15 22 PM" src="https://user-images.githubusercontent.com/2114494/215561384-6780373c-d578-4719-8a4c-a6f7f1c932e2.png">

I see errors like this while I'm running the app locally, but these helpful errors don't show up when I've deployed them to the server for internal testing. 

How can I make these errors show even after deploying to the server? 

Thanks! 🙂

**Comment by icyJoseph** [comment]
Next.js uses a packaged (made by them) called [react-dev-overlay](https://github.com/vercel/next.js/tree/canary/packages/react-dev-overlay), which does some aggressive error listening, not only through React Error boundaries but also other means, such as "listening" to unhandled rejections.

I do believe this is turned off on production builds, but you can try to implement and Error boundary yourself, to at least try to catch these on production. Error Boundaries are a React feature, which you can read about here https://reactjs.org/docs/error-boundaries.html.

**Comment by JoeFerrucci** [comment]
Got ya thanks. That's not really practical to do. I'm surprised and disappointed there's no way to run in dev mode while on a server, like you can on other frameworks (e.g., Rails). It's strange that you can't force the app show the debug info like it does while running locally.

**Reply by icyJoseph** [reply]
~I read that you can build with dev mode and start it as well. Like `NODE_ENV development npm run build` and sane for start. Haven't tried it myself though.~

Finally got around testing the above, as I had suspected, it looks like it works, but it doesn't. Sorry about that.

**Reply by icyJoseph** [reply]
Also, while I understand your feelings, it is important to learn to do error handling and put it in place yourself, rather than relying on any particular framework, moreover this kind of errors have to be caught at development time, it is rather scary to think of deploying this to production to test where we went wrong, at most we could prepare our code to deal with unexpected cases and report it back to us.

For example your error seems to be because of lack of `strictNullChecks` in your TypeScript configuration, or because some type casting that's causing your code to end up reading from `null`. 

That kind of problem ought to be handled first by your code, and then by error handlers and what not.  If the object that has the `itemId` property can be null, your types ought to reflect that.

**Reply by JoeFerrucci** [reply]
I fully agree with you. I’ve already shared that with my web engineer, which is why I came here looking for a solution to mitigate the issue while testing in alpha. It is indeed a null access case.
